drivers/auxdisplay/auxdisplay_hd44780.c

```diff
@@ -88,17 +88,76 @@ static void hd44780_pulse_enable_line(const struct device *dev)
 	k_sleep(K_NSEC(config->enable_line_fall_delay));
 }
 
-static void auxdisplay_hd44780_command(const struct device *dev, bool rs, uint8_t cmd,
-				       uint8_t mode)
+static inline void hd44780_set_rs_rw_lines(const struct device *dev, bool rs, bool rw)
+{
+	const struct auxdisplay_hd44780_config *const config = dev->config;
+
+	gpio_pin_set_dt(&config->rs_gpio, rs);
+	if (config->rw_gpio.port) {
+		gpio_pin_set_dt(&config->rw_gpio, rw);
+	}
+
+	k_sleep(K_NSEC(config->rs_line_delay));
+}
+
+static int hd44780_db_gpios_configure(const struct device *dev, uint8_t lsb_line,
+				      gpio_flags_t flags)
 {
+	const struct auxdisplay_hd44780_config *config = dev->config;
+	int rc;
+
+	for (int line = 7; line >= lsb_line; --line) {
+		rc = gpio_pin_configure_dt(&config->db_gpios[line], flags);
+		if (rc < 0) {
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+static void auxdisplay_hd44780_command(const struct device *dev, bool rs,
+				       uint8_t cmd, uint8_t mode)
+{
+	int rc;
 	const struct auxdisplay_hd44780_config *config = dev->config;
 	int8_t i = 7;
 	const int lsb_line = (mode == AUXDISPLAY_HD44780_MODE_8_BIT) ? 0 : 4;
 	int ncommands = (mode == AUXDISPLAY_HD44780_MODE_4_BIT) ? 2 : 1;
+	const bool check_busy_flag = (!config->rw_gpio.port ||
+				      (mode == AUXDISPLAY_HD44780_MODE_4_BIT_ONCE)) ?
+				      false : true;
 
-	gpio_pin_set_dt(&config->rs_gpio, rs);
-	k_sleep(K_NSEC(config->rs_line_delay));
+	if (check_busy_flag) {
+		bool busy;
+
+		rc = hd44780_db_gpios_configure(dev, lsb_line, GPIO_INPUT | GPIO_PULL_DOWN);
+		if (rc < 0) {
+			LOG_ERR("configuration of db-gpios as inputs failed: %d", rc);
```

Comment:
Configuration (with capital C)

Reply:
Agreed.

Reply:
@kuliga comments like this one is why I pinged you :) I think maybe you didn't push all the updates you intended to?

Reply:
In any case, when ready, you may re-request reviews from the people who left comments from the GitHub UI to make it clear it's ready to be reviewed again
Cheers!

Reply:
Oh, I missed that. Thanks for noticing.